### DISCUSSION TITLE
Best practice for updating: should I wait before I update esphome on NS panel?

### DISCUSSION THREAD

**Discussion by justinotherguy** [Blackymas/NSPanel_HA_Blueprint#839] [General]
Hi there,

really love your work - much appreciated!

Every time I am offered an update for esphome, I never know whether to update my ns panel, too or not.
I've done it without an issue - and I needed to update the blueprint and tft another time.

What's the best way to go about?

Regards, J.

**Comment by Doughboy68** [comment]
You are generally ok with performing esphome updates. It will most of the time pull same code down from the main branch and just recompile with new esphome libraries. However, if you wait a very long time between updates, and are not keeping up with latest versions of NSPanel Blueprint, then the latest main branch will most likely require a new blueprint file and tft. Having said that, ESPhome updates however come out more often than NSPanel updates so if you keep on top of them you will probably not experience this issue. What I would do is wait a day or so after ESPHome update and monitor this discussion area to make sure there are no breaking changes... eg. last month there was one but was fixed almost immediately. I just did all my panels recently with no issues.

**Reply by justinotherguy** [reply]
excellent, thank you! :)
